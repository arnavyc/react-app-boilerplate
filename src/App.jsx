/**
 * Copyright 2021 arnavyc <arnavyc@outlook.com>
 * SPDX-License-Identifier: 0BSD
 */

import React from 'react'

function App() {
  return (
    <div>
      <p>Hello, World</p>
    </div>
  )
}

export default App
