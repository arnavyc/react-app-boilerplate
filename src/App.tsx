/**
 * Copyright 2021 arnavyc <arnavyc@outlook.com>
 * SPDX-License-Identifier: 0BSD
 */

import React, { type ReactNode } from 'react'

function App(): ReactNode {
  return (
    <div>
      <p>Hello, World</p>
    </div>
  )
}

export default App
