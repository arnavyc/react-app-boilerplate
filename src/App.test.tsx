/**
 * SPDX-FileCopyrightText: 2021 arnavyc
 *
 * SPDX-License-Identifier: 0BSD
 */

import React from 'react'
import ReactDOMClient from 'react-dom/client'
import renderer from 'react-test-renderer'
import App from './App'

describe('<App /> component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    const root = ReactDOMClient.createRoot(div)
    root.render(<App />)
    root.unmount()
  })

  it('matches snapshot', () => {
    const tree = renderer.create(<App />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
