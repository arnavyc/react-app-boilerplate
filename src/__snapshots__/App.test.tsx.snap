// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> component matches snapshot 1`] = `
<div>
  <p>
    Hello, World
  </p>
</div>
`;
